### PR TITLE
Add redaction example

### PR DESCRIPTION
**Pull request by dashpole**
This example shows how to use the redaction processor to remove credit card numbers.  It is based on the example config used in the redaction processor documentation: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/redactionprocessor.  It uses the otlpjsonfile receiver to generate an input span to show how the feature works.

I removed the try it out section as the json file receiver doesn't appear to work at all...

### CODE REVIEW

File: deploy/gke/redaction/README.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+card numbers.
+
+```
+kubectl create configmap testdata --from-file=./testdata.json -n $OTEL_NAMESPACE
+```
+
+
+### Create the Deployment
+
+When you built your collector image, the file [`manifest.yaml`](manifest.yaml)
```
Comment (damemi) [inline deploy/gke/redaction/README.md]: should this link to the collector-image builder steps?

File: deploy/gke/redaction/README.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+was automatically updated to reference your image name. Create this manifest in your cluster with:
+
+```
+cd deploy/gke/simple/
+kubectl apply -f manifest.yaml -n $OTEL_NAMESPACE
+```
+
+### Try it out!
+
+Take a look at [`testdata.json`](testdata.json`).  It contains an attribute
```
Comment (damemi) [inline deploy/gke/redaction/README.md]:
```suggestion
Take a look at [`testdata.json`](testdata.json).  It contains an attribute
```

File: deploy/gke/redaction/README.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+#### Remove Kubernetes resources
+
+```
+kubectl delete -f manifest.yaml -n $OTEL_NAMESPACE
+
+kubectl delete namespace $OTEL_NAMESPACE
+```
+
+##### (Cleanup) Remove Artifact Registry permissions from default service account
```
Comment (damemi) [inline deploy/gke/redaction/README.md]: now we know for sure external users shouldn't need to do this, so we can drop this section

File: deploy/gke/redaction/README.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+Take a look at [`testdata.json`](testdata.json`).  It contains an attribute
+`credit.card.number=visa credit card number: 4111111111111111` which matches
+our redaction regexp filter in [`otel-config.yaml`](otel-config.yaml):
+`4[0-9]{12}(?:[0-9]{3})?`.
+
+Check the collector deployment's logs to see the attribute now that it has been redacted:
+
+```
+kubectl logs deploy/otel-collector -n $OTEL_NAMESPACE | grep "visa credit card number"
```
Comment (damemi) [inline deploy/gke/redaction/README.md]: a sample of the actual redacted lines would be nice here to compare the user's output against

**Review by damemi — COMMENTED** [review]
only thing that needs to be fixed is the broken link, the rest is just suggestions

File: deploy/gke/redaction/README.md (hunk 2 above)
Comment (dashpole) [reply deploy/gke/redaction/README.md]: removed this section

File: deploy/gke/redaction/README.md (hunk 4 above)
Comment (dashpole) [reply deploy/gke/redaction/README.md]: removed this section

File: deploy/gke/redaction/README.md (hunk 3 above)
Comment (dashpole) [reply deploy/gke/redaction/README.md]: removed

File: deploy/gke/redaction/README.md (hunk 1 above)
Comment (dashpole) [reply deploy/gke/redaction/README.md]: removed this sentence

**Review by damemi — APPROVED** [review]

**Pull request merged**